kpatch-build/create-klp-module.c

```diff
@@ -208,8 +239,23 @@ static void create_klp_relasecs_and_syms(struct kpatch_elf *kelf, struct section
 		ALLOC_LINK(rela, &klp_relasec->relas);
 		rela->sym = sym;
 		rela->type = krelas[index].type;
-		rela->offset = krelas[index].offset;
-		rela->addend = krelas[index].addend;
+		if (!strcmp(base->name, ".toc"))
+			rela->offset = toc_offset;
+		else
+			rela->offset = krelas[index].offset;
```

Comment:
The reason, why it been handled here is. When more two or more objects files are linked together, their .toc enteries are added to the .toc section of kpatch-output object and there is no fool proof way to predictable the arrangement of entries in the combined .toc section. From ```kmod/patch/Makefile``` file ```kpatch-$(KPATCH_NAME)-objs += patch-hook.o kpatch.lds output.o``` . When .toc section entries from output.o and patch-hook.o are merged, the arrangement of the combined .toc is not guaranteed to map the index based offset. I have added a comment with an example, explaining the re-arrangement at line 177 explaining this re-arrangement. I guess, it better to move it here.